spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -1072,9 +1072,61 @@ struct ProductionVisitor
         pushBuilder(body);
         body->addExpression(builder::decrementPostfix(builder::id("__i")));
 
-        auto stop = parseProduction(p.body());
-        auto b = builder()->addIf(stop);
-        b->addBreak();
+        builder()->addComment("NOPE production::Counter");
+
+        auto parse = [&]() {
+            auto stop = parseProduction(p.body());
+            auto b = builder()->addIf(stop);
+            b->addBreak();
+        };
+
+        // The container creating this counter was marked `&synchronize`. Allow
+        // any container element to fail parsing and be skipped. This means
+        // that if `n` elements where requested and one element fails to parse,
+        // we will return `n-1` elements.
+        if ( auto c = p.body().meta().container(); c && AttributeSet::find(c->attributes(), "&synchronize") ) {
```

Comment:
As discussed offline, this overloads `&synchronize` to mean both external and internal list sync, e.g.,

```.ruby
type X = unit {
    : /ABC/;
    ys: Y[] &synchronize;  # Triggers both internal and external sync.
};

type Y = unit { : /Y/; };
```

Let's instead trigger internal sync with an attribute on the list element type, e.g,.

```.ruby
ys: (Y &synchronize)[];
```